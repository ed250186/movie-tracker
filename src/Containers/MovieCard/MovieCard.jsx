import React, { Component } from "react";
import "./MovieCard.scss";
import PropTypes from "prop-types";
import active from "../../images/star-active.png";
import inactive from "../../images/star-inactive.png";
import { connect } from "react-redux";
import {
  addNewFavorite,
  fetchFavoriteMovies,
  deleteFav
} from "../../apiCalls/apiCalls";
import {
  addFavoriteMovie,
  deleteFavoriteMovie,
  getFavoriteMovies
} from "../../actions/favoriteAction";

export class MovieCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      favorited: false,
      error: "",
      movieInfo: false
    };
  }

  addFavorites = () => {
    const {
      id,
      userId = this.props.login.id,
      title,
      posterPath,
      releaseDate,
      voteAverage,
      overview
    } = this.props;
    const addFav = addNewFavorite(
      id,
      userId,
      title,
      posterPath,
      releaseDate,
      voteAverage,
      overview
    );
    this.props.addFavoriteMovie(addFav);

    // if (addFav) {
    //   this.props.addFavoriteMovie(addFav);
    // } else {
    //   this.setState({ error: "Please login to be able to favorite a movie" });
    // }
    // if (this.props.login.loggedIn) {
    //   this.toggleFavorite();
    // }
  };

  // toggleFavorite = (obj, url, func, err) => {

  //     this.props.addFavoriteMovie(favMovie.data)

  //   if (!this.state.favorited) {
  //     this.setState({ favorited: true });
  //   } else {
  //     this.setState({ favorited: false });
  //     this.deleteFavoriteMovie(this.props.login.id, this.props.id)
  //   }
  // // }

  toggleFavorite = async (e) => {
    e.preventDefault();
    
    if (!this.props.login.loggedIn) {
      this.setState({ error: "Please login to be able to favorite a movie" });
    } else {
      const userId = this.props.login.id;
      let addFavorite = await fetchFavoriteMovies(userId);
      this.props.getFavoriteMovies(addFavorite);

      const { favorites, movies } = this.props;
        return favorites.find(fav => {
          if (!fav.movie_id == movies.id) {
            // this.addFavorites();
            // this.setState({ favorited: true });
            console.log('added')
          } else {
            // this.setState({ favorited: false });
            console.log(movies)
            console.log('hiiiiiiii')
          }
        });
      };
      console.log(this.props.favorites);
      // if(this.props.favorites){}
  };
    //   this.deleteFavoriteMovie(this.props.login.id, this.props.id);

  deleteFavoriteMovie = async (userId, movieId) => {
    const movieData = { user_id: userId, movie_id: movieId };
    await this.props.deleteFavoriteMovie(movieData);
    console.log("hiiiiiiiiiii");
    // await this.props.deleteFavoriteMovie(movieData)
  };

  toggleView = view => {
    this.setState({ movieInfo: view });
  };

  infoCard = () => {
    const { title, backDropPath, voteAverage, overview } = this.props;
    return (
      <article className="infoCard">
        <img
          src={backDropPath}
          alt={title + "background image"}
          onClick={() => this.toggleView(false)}
        />
        <p className="title">
          {title} | {voteAverage}/10
        </p>
        <p className="overview">{overview}</p>
      </article>
    );
  };

  render() {
    const { title, posterPath, releaseDate } = this.props;
    return (
      <article className="movieCard">
        {this.state.error === "" ? <p /> : <p>{this.state.error}</p>}
        {this.state.movieInfo && this.infoCard()}
        <img
          src={this.state.favorited ? active : inactive}
          alt="inactive"
          className="inactive"
          onClick={e => this.toggleFavorite(e)}
        />
        <img
          className="card-img"
          src={posterPath}
          alt={`${title} poster`}
          onClick={() => this.toggleView(true)}
        />
        <p className="title">{title}</p>
        <p className="year">{releaseDate}</p>
      </article>
    );
  }
}

const mapStateToProps = state => ({
  movies: state.movies,
  login: state.login,
  favorites: state.favorites,
  userId: state.userId
});

const mapDispatchToProps = dispatch => ({
  addFavoriteMovie: movie => dispatch(addFavoriteMovie(movie)),
  deleteFavoriteMovie: movie => dispatch(deleteFavoriteMovie(movie)),
  getFavoriteMovies: movie => dispatch(getFavoriteMovies(movie))
});

MovieCard.propTypes = {
  posterPath: PropTypes.string,
  title: PropTypes.string,
  releaseDate: PropTypes.string
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MovieCard);
