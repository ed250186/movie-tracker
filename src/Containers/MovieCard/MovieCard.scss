
.movieCard {
  height: 250px;
  width: 140px;
  margin: 30px;
  position: relative;
  .card-img{
    height: auto;
    width: 140px;
    box-shadow: 2px 2px 10px black;
  }
  .inactive{
    position: absolute;
    z-index: 1;
    height: auto;
    width: 40px;
    left: 100px;
  }
  .title{
    color: #F4F8FA;
    margin: 5px 0 5px 0;
  }
  .year{
    color: #49597B;
  }
}

.infoCard {
  width: 400%;
  position: absolute;
  z-index: 5;
  background: #314467;
  top: -5%;
  left: -5%;
  box-shadow: 2px 2px 10px black;
  img {
    width: 90%;
    height: auto;
    margin: 5%;
  }
  .title {
    font-size: 1.3rem;
  }
  .overview{
    color: white;
    font-size: .9rem;
  }
}
