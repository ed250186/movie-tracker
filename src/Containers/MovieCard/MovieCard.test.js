import React from 'react';
import {shallow} from 'enzyme';
import {MovieCard} from './MovieCard';

describe('App', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<MovieCard/>);
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

})