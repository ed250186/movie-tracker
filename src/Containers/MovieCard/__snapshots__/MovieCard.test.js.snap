// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match snapshot 1`] = `
<article
  className="movieCard"
>
  <p />
  <img
    alt="inactive"
    className="inactive"
    onClick={[Function]}
    src="star-inactive.png"
  />
  <img
    alt="undefined poster"
    className="card-img"
    onClick={[Function]}
  />
  <p
    className="title"
  />
  <p
    className="year"
  />
</article>
`;
